features/step_definitions/wishlist_steps.rb

```diff
@@ -1,64 +1,45 @@
+  @course = @wishlist_item.course
 end
 
-Given(/^I do not have the course ([A-Z]+) (\d+) in my wishlist$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number).first
-  if course
-    # if there's no course, there's definitely no wishlist item
-    WishlistItem.destroy_all(course: course, user: @current_user)
-  end
-end
-
-Given(/^My wishlist is empty$/) do
-  WishlistItem.destroy_all(user: @current_user)
-  expect(WishlistItem.where(user: @current_user).to_a).to eq []
+Given(/^I have notifications turned (on|off) for that course$/) do |m|
+  status = m == 'on'
+  @wishlist_item.update_attributes!(notify: status)
 end
 
-Given(/^I do not have notifications turned on for ([A-Z]+) (\d+)$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number).first
-  WishlistItem.where(course: course, user: @current_user).update_all(notify: false)
-  expect(WishlistItem.where(course: course, user: @current_user, notify: true).to_a).to eq []
+When(/^I view my wishlist$/) do
+  step 'I visit "/wishlist"'
 end
 
-Given(/^I have notifications turned on for ([A-Z]+) (\d+)$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number).first
-  WishlistItem.where(course: course, user: @current_user).update_all(notify: true)
-  expect(WishlistItem.where(course: course, user: @current_user, notify: false).to_a).to eq []
+When(/^I select a course via wishlist search$/) do
+  @course = Course.first || FactoryGirl.create(:course)
+  step "I fill in \"course_title\" with \"#{@course.to_param}\""
+  step 'I click the hidden button "#submit"'
 end
 
-Then(/^I should have the course ([A-Z]+) (\d+) in my wishlist$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number).first
-  expect(course).to_not be nil
-  expect(WishlistItem.exists?(course: course, user: @current_user)).to be true
+When(/^I add it to my wishlist$/) do
+  step 'I click the link "Add to my wishlist"'
 end
 
-Then(/^I should not have the course ([A-Z]+) (\d+) in my wishlist$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number).first
-  # If the course doesn't exist it's definitely not in the wishlist
-  expect(course && WishlistItem.exists?(course: course, user: @current_user)).to be false
+When(/^I remove it from my wishlist$/) do
+  step 'I click the link "Remove from my wishlist"'
 end
 
-Then(/^I should have notifications turned on for ([A-Z]+) (\d+)$/) do |course_dept, course_number|
-  course = Course.where(department: course_dept, course_number: course_number)
-  expect(WishlistItem.where(course: course, user: @current_user).first.notify).to be true
+When(/^I turn (on|off) notifications$/) do |m|
+  step "I click the link \"Turn #{m} notifications\""
```

Comment:
Either remove all the "I click" steps or just use the original step names. Related to my previous comment.